### ISSUE TITLE
install (git clone) failures

### ISSUE THREAD

**Issue by linuxcaffe** [jubnzv/qtask#3]
In trying to install qtask, after successfully apt installing the requirements, I run into these errors;

`➜  tmp git clone --recurse-submodules https://github.com/jubnzv/qtask.git qtask
Cloning into 'qtask'...
remote: Enumerating objects: 386, done.
remote: Counting objects: 100% (386/386), done.
remote: Compressing objects: 100% (249/249), done.
remote: Total 386 (delta 252), reused 263 (delta 130), pack-reused 0
Receiving objects: 100% (386/386), 255.17 KiB | 620.00 KiB/s, done.
Resolving deltas: 100% (252/252), done.
Submodule 'src/third-party/singleapplication' (git@github.com:itay-grudev/SingleApplication.git) registered for path 'src/third-party/singleapplication'
Cloning into '/home/djp/tmp/qtask/src/third-party/singleapplication'...
The authenticity of host 'github.com (140.82.114.3)' can't be established.
RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
Are you sure you want to continue connecting (yes/no)? y
Please type 'yes' or 'no': yes
Warning: Permanently added 'github.com,140.82.114.3' (RSA) to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:itay-grudev/SingleApplication.git' into submodule path '/home/djp/tmp/qtask/src/third-party/singleapplication' failed
Failed to clone 'src/third-party/singleapplication'. Retry scheduled
Cloning into '/home/djp/tmp/qtask/src/third-party/singleapplication'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:itay-grudev/SingleApplication.git' into submodule path '/home/djp/tmp/qtask/src/third-party/singleapplication' failed
Failed to clone 'src/third-party/singleapplication' a second time, aborting
`

what went wrong?

**Comment by jubnzv** [comment]
@linuxcaffe Hi,

The problem is that you have not configured (or incorrectly configured) an [SSH connection to GitHub](https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh).

The quick and dirty solution will be:
```bash
cd /home/djp/tmp/qtask/
rm -rf src/third-party/singleapplication
git clone https://github.com/itay-grudev/SingleApplication src/third-party/singleapplication
```

Then you can continue building `qtask` according to the README.

If you want to properly set up SSH connection with GitHub, I recommend reading [their troubleshooting guide](https://docs.github.com/en/github/authenticating-to-github/testing-your-ssh-connection) first. Probably you will need to add your SSH key to the GitHub account. Then `git clone` with submodules will work correctly.

**Comment by linuxcaffe** [comment]
Well thanks for the quick response and the interesting project. I'm sure you're right, i need to set up SSH right with GitHub, I'll do that and try again.. sadly, the fairly-quick and way-too-dirty approach failed :)

`[ 36%] Building CXX object CMakeFiles/qtask.dir/src/trayicon.cpp.o
[ 40%] Building CXX object CMakeFiles/qtask.dir/src/taskdialog.cpp.o
/home/djp/tmp/qtask/src/taskdialog.cpp: In member function ‘void TaskDialog::initUI()’:
/home/djp/tmp/qtask/src/taskdialog.cpp:92:56: error: ‘class QDate’ has no member named ‘startOfDay’
     m_task_sched->setMinimumDateTime(QDate(1980, 1, 2).startOfDay());
                                                        ^~~~~~~~~~
/home/djp/tmp/qtask/src/taskdialog.cpp:93:56: error: ‘class QDate’ has no member named ‘startOfDay’
     m_task_sched->setMaximumDateTime(QDate(2038, 1, 1).startOfDay());
                                                        ^~~~~~~~~~
/home/djp/tmp/qtask/src/taskdialog.cpp:94:52: error: ‘class QDate’ has no member named ‘startOfDay’
     m_task_sched->setDateTime(QDate::currentDate().startOfDay().addDays(1));
                                                    ^~~~~~~~~~
/home/djp/tmp/qtask/src/taskdialog.cpp:98:54: error: ‘class QDate’ has no member named ‘startOfDay’
     m_task_due->setMinimumDateTime(QDate(1980, 1, 2).startOfDay());
                                                      ^~~~~~~~~~
/home/djp/tmp/qtask/src/taskdialog.cpp:99:54: error: ‘class QDate’ has no member named ‘startOfDay’
     m_task_due->setMaximumDateTime(QDate(2038, 1, 1).startOfDay());
                                                      ^~~~~~~~~~
/home/djp/tmp/qtask/src/taskdialog.cpp:100:50: error: ‘class QDate’ has no member named ‘startOfDay’
     m_task_due->setDateTime(QDate::currentDate().startOfDay().addDays(5));
                                                  ^~~~~~~~~~
/home/djp/tmp/qtask/src/taskdialog.cpp:104:55: error: ‘class QDate’ has no member named ‘startOfDay’
     m_task_wait->setMinimumDateTime(QDate(1980, 1, 2).startOfDay());
                                                       ^~~~~~~~~~
/home/djp/tmp/qtask/src/taskdialog.cpp:105:55: error: ‘class QDate’ has no member named ‘startOfDay’
     m_task_wait->setMaximumDateTime(QDate(2038, 1, 1).startOfDay());
                                                       ^~~~~~~~~~
/home/djp/tmp/qtask/src/taskdialog.cpp:106:51: error: ‘class QDate’ has no member named ‘startOfDay’
     m_task_wait->setDateTime(QDate::currentDate().startOfDay().addDays(5));
                                                   ^~~~~~~~~~
CMakeFiles/qtask.dir/build.make:182: recipe for target 'CMakeFiles/qtask.dir/src/taskdialog.cpp.o' failed
make[2]: *** [CMakeFiles/qtask.dir/src/taskdialog.cpp.o] Error 1
CMakeFiles/Makefile2:68: recipe for target 'CMakeFiles/qtask.dir/all' failed
make[1]: *** [CMakeFiles/qtask.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
`

**Comment by jubnzv** [comment]
What version of Qt are you using?
You need at least 5.14.

**Comment by linuxcaffe** [comment]
looks like qt* is at 5.9.5+dfsg-0ub

**Comment by jubnzv** [comment]
Got it. I'll take a look how we can add support for Qt 5.9 this week. I created this issue: #4.
The implementation will be a bit tricky, since some useful Qt functions like markdown support in `QTextDocument` are not available in 5.9.

**Comment by jubnzv** [comment]
@linuxcaffe I added Qt 5.9 support in #5.

I think, all the features will work, expect for the markdown support in the tasks list. At least [CI builds](https://github.com/jubnzv/qtask/runs/1800004583) with Qt 5.9 were successfull.
